### PR TITLE
Make WebElement can be used as param of execScript

### PR DESCRIPTION
**Pull request by tinyshrimp**
Make WebElement can be used as a parameter of execScript method.
Related detail information, please see  
- [WebDriver Elements Section](https://www.w3.org/TR/webdriver/#elements) on W3C
- Source code of org.openqa.selenium.remote.internal.WebElementToJsonConverter class of Java

### CODE REVIEW

File: remote.go [hunk 1]
```diff
@@ -1088,6 +1096,13 @@ func (elem *remoteWE) CSSProperty(name string) (string, error) {
 	return elem.parent.stringCommand(urlTemplate)
 }
 
+func (elem *remoteWE) toMap() map[string]string {
+	element := make(map[string]string)
+	element["ELEMENT"] = elem.id
+	element["element-6066-11e4-a52e-4f735466cecf"] = elem.id
```
Comment (minusnine) [inline remote.go]: Simplify this to:

```go
return map[string]string{
  "ELEMENT": elem.id,
  "element-6066-11e4-a52e-4f735466cecf": elem.id,
}
```

File: remote.go [hunk 2]
```diff
@@ -844,6 +844,14 @@ func (wd *remoteWD) SetAlertText(text string) error {
 }
 
 func (wd *remoteWD) execScriptRaw(script string, args []interface{}, suffix string) ([]byte, error) {
+	for index, value := range args {
+		if ori, ok := value.(*remoteWE); ok {
+			args[index] = ori.toMap()
+		} else if ori, ok := value.(remoteWE); ok {
+			args[index] = ori.toMap()
+		}
+	}
```
Comment (minusnine) [inline remote.go]: Please add a test. The test should demonstrate that some trivial script can make effective use of a WebElement passed in this way.

**Comment by minusnine** [comment]
Also: thank you for your contribution!

File: remote.go (hunk 2 above)
Comment (tinyshrimp) [inline remote.go]: OK, I will add test for this change.

File: remote.go (hunk 1 above)
Comment (tinyshrimp) [inline remote.go]: Got it.

**Comment by tinyshrimp** [comment]
@minusnine 
the test cases have been added, and all passed the test in travis.
Please review it. Thank you.

File: remote.go [hunk 3]
```diff
@@ -781,10 +781,23 @@ func (wd *remoteWD) SetAlertText(text string) error {
...
+				args[index] = ori.toMap()
+			}
+		}
+	}
+
 	data, err := json.Marshal(map[string]interface{}{
 		"script": script,
 		"args":   args,
 	})
+
```
Comment (minusnine) [inline remote.go]: nit: Remove this empty line between the function call and its error handling.

File: remote.go [hunk 4]
```diff
@@ -781,10 +781,23 @@ func (wd *remoteWD) SetAlertText(text string) error {
 }
 
 func (wd *remoteWD) execScriptRaw(script string, args []interface{}, suffix string) ([]byte, error) {
+	if args == nil {
+		args = make([]interface{}, 0)
+	} else {
+		for index, value := range args {
+			if ori, ok := value.(*remoteWE); ok {
+				args[index] = ori.toMap()
+			} else if ori, ok := value.(remoteWE); ok {
```
Comment (minusnine) [inline remote.go]: Will the value ever be a non-pointer? In other words, will this branch ever execute?

I think we only ever emit the *remoteWE type as WebElements, so this branch can be removed.

File: remote_test.go [hunk 5]
```diff
@@ -758,6 +760,71 @@ func testExecuteScript(t *testing.T, c config) {
...
+	wd := newRemote(t, c)
+	defer quitRemote(t, wd)
+
+	if err := wd.Get(serverURL); err != nil {
+		t.Fatalf("wd.Get(%q) returned error: %v", serverURL, err)
+	}
+
+	script := "return document.readyState"
+	_, err := wd.ExecuteScript(script, nil)
+	if err != nil {
```
Comment (minusnine) [inline remote_test.go]: nit: Combine this with the previous line:

```go
if _, err := wd.ExecuteScript(...); err != nil {
...
}
```

File: remote_test.go [hunk 6]
```diff
@@ -758,6 +760,71 @@ func testExecuteScript(t *testing.T, c config) {
...
+
+	input, err := wd.FindElement(ByName, "q")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := input.SendKeys("golang"); err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(500 * time.Millisecond)
```
Comment (minusnine) [inline remote_test.go]: Is this sleep necessary? Remove it if it is not.

File: remote_test.go [hunk 7]
```diff
@@ -758,6 +760,71 @@ func testExecuteScript(t *testing.T, c config) {
...
+
+	we, err := wd.FindElement(ByXPATH, "//input[@type='submit']")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	script := "arguments[0].click()"
+	args := []interface{}{we}
+	_, err = wd.ExecuteScript(script, args)
+	if err != nil {
```
Comment (minusnine) [inline remote_test.go]: Combine with the previous line.

**Comment by tinyshrimp** [comment]
@minusnine
Modified with your comments.
Thank you.

**Comment by minusnine** [comment]
Thanks again!

**Pull request merged**